### ISSUE TITLE
Help for publishing without server

### ISSUE THREAD

**Issue by psamim** [kriasoft/react-static-boilerplate#163]
Hi,

Is it possible to publish the Website in way to use it without any server and use just the browser? Currently when I build (`node run build --release`), the resulting files are static but need / as their path (a static file server is needed).

I know the _issues_ are just for reporting issues but I would appreciate if someone could help.

Thanks,
Samim

**Comment by frenzzy** [comment]
Have you seen https://github.com/kriasoft/react-starter-kit/pull/642 ? Is this exactly what you need?

**Comment by psamim** [comment]
Not exactly, the above generates files for every route. I meant having one index.html file that can be opened with browser without having any static file server. I have built my project with react-static-boilerplate and I do not have the knowledge to switch to react-starter-kit. Now when I build my project I can use it with a server, but not I cannot use it by directly opening the index.html file with a browser. I would be very grateful if you could give me some clues what to do.

**Comment by psamim** [comment]
@frenzzy 
Maybe the problem is with `history`, I get the following in the console:

```
Uncaught SecurityError: Failed to execute 'pushState' on 'History': A history state object with URL 'file:///' cannot be created in a document with origin 'null' and URL 'file:///home/...````

**Comment by koistya** [comment]
@psamim even if you open `index.html` in a browser, it references images and .js files, how the HTML page would load them without a server? You could try to embed everything into application bundle (main.js) and then embed that JavaScript code into your HTML page, so that the output folder would contain a single index.html file, if you do so.. it may work without a server.

**Comment by psamim** [comment]
Not exactly, the above generates files for every route. I meant having one
index.html file that can be opened with browser without having any static
file server. I have built my project with react-static-boilerplate and I do
not have the knowledge to switch to react-starter-kit. Now when I build my
project I can use it with a server, but not I cannot use it by directly
opening the index.html file with a browser. I would be very grateful if you
could give me some clues what to do.

**Comment by greg234234** [comment]
I keep finding this thread when searching for why my local copy doesn't produce html files, only js files + images, and when I deploy it says Page Not Found on the deployed firebase servers.  Is this releated?
